File: app/src/main/java/com/robotnec/budget/core/domain/operation/Income.java
package com.robotnec.budget.core.domain.operation;

import com.robotnec.budget.core.domain.Account;

import lombok.Data;

/**
 * @author zak <zak@swingpulse.com>
 */
@Data
// example
public class Income implements Operation {

    private long date;
    private Account account;

    @Override
    public boolean execute(OperationReceiver receiver) {
        return receiver.receive(this);
    }
}

File: app/src/main/java/com/robotnec/budget/core/domain/operation/Income.kt
package com.robotnec.budget.core.domain.operation

import com.robotnec.budget.core.domain.Account

/**
 * @author zak zak@swingpulse.com>
 */
// example
class Income(val date: Long,
             val account: Account) : Operation {

    override fun execute(receiver: OperationReceiver): Boolean {
        return receiver.receive(this)
    }
}
